### PR TITLE
Add OneDrive iframe file selector

### PR DESCRIPTION
**Pull request by CronkCode**
# Add OneDrive iframe file selector

## Details on the issue fix or feature implementation

- Adds OneDrive iframe file selector to UserPortal
- Updated file upload experience to match mockup
- Create OneDrive endpoints and service in the Core API
- Adds file store connector methods

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [x]  This PR needs to be cherry-picked into at least one release branch
- [ ]  I have included unit tests for the issue/feature
- [x]  I have included inline docs for my changes, where applicable
- [x]  I have successfully run a local build
- [x]  I have provided the required update scripts, where applicable
- [x]  I have updated relevant docs, where applicable

### CODE REVIEW

File: src/dotnet/Core/Interfaces/IOneDriveService.cs
Comment (ciprianjichici) [inline src/dotnet/Core/Interfaces/IOneDriveService.cs:1]: We should use a consistent naming approach. Preferred is `OneDriveWorkSchool`.

File: src/dotnet/Core/Models/OneDriveItem.cs
Comment (ciprianjichici) [inline src/dotnet/Core/Models/OneDriveItem.cs:1]: We should use a consistent naming approach. Preferred is `OneDriveWorkSchool`.

File: src/dotnet/Core/Services/OneDriveService.cs
Comment (ciprianjichici) [inline src/dotnet/Core/Services/OneDriveService.cs:1]: We should use a consistent naming approach. Preferred is `OneDriveWorkSchool`.

File: src/dotnet/Core/Models/OneDriveItem.cs
Comment (ciprianjichici) [inline src/dotnet/Core/Models/OneDriveItem.cs:1]: What is the logic behind isolating the mime type only in a separate class, while the rest of the properties (object id and name) are not?

File: src/ui/UserPortal/stores/authStore.ts
Comment (ciprianjichici) [inline src/ui/UserPortal/stores/authStore.ts:1]: We should not hardcode scope values.

Comment (ciprianjichici) [inline src/ui/UserPortal/stores/authStore.ts:1]: We should use a consistent naming approach. Preferred is `OneDriveWorkSchool`.

File: src/dotnet/Common/Templates/AppConfigurationKeys.cs
[the remark this answers is not in the document]
Comment (joelhulen) [reply src/dotnet/Common/Templates/AppConfigurationKeys.cs:1]: I'm not sure where this came from. The UX is currently using the `APIEndpointConfiguration` resource, not this.

**Review by ciprianjichici — APPROVED** [review]

**Pull request merged**